### ISSUE TITLE
Convert `dependencies.txt` files into `dependencies.sh` files

### ISSUE THREAD

**Issue by postmodern** [postmodern/ruby-install#477]
Replace the `dependencies.txt` file with shell script `dependencies.sh` files that better allow for dynamically defining package manager dependencies (ex: `openssl@1.1` vs. `openssl@3` or `jemalloc`).

### Example Code

```shell
case "$package_manager" in
    apt)    dependencies=(...) ;;
    brew)
        dependencies=(...)
        
        case "$ruby_version" in
            2.*|3.0.*)    dependencies+=("openssl@1.1") ;;
            *)            dependencies+=("openssl@3") ;;
        esac
        ;;        
    ...
esac
```

The `install_optional_deps` and the `fetch` functions could also be removed. The `load_dependencies` function could be replaced with a simple `source "$ruby/dependencies.sh"` line.

**Comment by ParadoxV5** [comment]
> No need to install Bison to build Ruby from source code anymore.
> ⸺ https://github.com/ruby/ruby/blob/v3_3_0/NEWS.md#parser

Similarly for `readline`, but it’s probably still used by the *default gem* <q>readline 0.0.4</q>

> We no longer need to install libraries like `libreadline` or `libedit`.
> ⸺ https://github.com/ruby/ruby/blob/v3_3_0/NEWS.md#stdlib-compatibility-issues

**Comment by postmodern** [comment]
Implemented by 25d4780 in the 0.10.0 branch.
